blog: cargo-information pre-RFC

hi-rustin opened the pull request.

hi-rustin commented:
cc: @epage Would you like to review the document before it's posted on the internal forum? If you're short on time, you can also provide feedback after publication. Thanks in advance.

epage commented:
[rendered](https://github.com/hi-rustin/hi-rustin.rs/blob/rustin-patch-blog-cargo-information/_posts/2024-02-01-pre-RFC-cargo-information.md)

epage commented:
I went ahead and did #109 real quick but overall, I think the question is what is your goal.
- This is written for your blog but generally things like this would be posted to Internals
- This is titled as a Pre-RFC but we don't generally use RFCs for subcommands.  Instead, we use a form of a change proposal.
  - Examples: https://github.com/rust-lang/cargo/pull/11099, https://github.com/rust-lang/cargo/pull/10472
  - We should probably document this
- When I've worked to merge commands, I focused on getting feedback on the unknown or uncertain areas from Internals
  - https://internals.rust-lang.org/t/feedback-on-cargo-upgrade-to-prepare-it-for-merging/17101/130
  - https://internals.rust-lang.org/t/feedback-on-cargo-add-before-its-merged/16024

epage commented:
> I want to focus on getting feedback on the unknown or uncertain designs and issues. I should emphasize the goal at the beginning or I may have too many details in this blog. What do you think? Should we put that many details in it or not?

Would Internals be a better place, than your blog?

So the goal (and now title) is about feedback but the structure doesn't show that.

For `cargo add` and `cargo upgade`,
- Opened with a very brief piece of context (related issue, that I'm looking for feedback)
- Provided a brief summary of what is done today titled "background".  The title is important.  "Summary" makes it sound like it'll be about your post.  "Background" makes it clear that you are providing context for the questions you want answered.
- The background section is brief.  I did this by focusing on the screenshot and then calling out some non-obvious points
- The remainder should be focused on the areas of concern with what context you can provide

hi-rustin commented:
> Would Internals be a better place, than your blog?

Yes, I will post it in the Internals. I utilize this blog as a platform to review the code(content) before publishing it.

hi-rustin commented:
@epage I updated it again. I deleted the scenario demos. Mentioned some background on the command. Also emphasized that I want to discuss those issues that are still undecided. Do you think this is ok for now?

hi-rustin commented:
I am going to merge this PR. I will publish it on internals later and if you have anything want to add I think you can comment on the internals forum.

Thanks for your review!

hi-rustin commented:
Live now: https://internals.rust-lang.org/t/feedback-on-cargo-info-to-prepare-it-for-merging/20369
